MobileApplication/MobileApplication/ViewModels/NewCategoryViewModel.cs

```diff
@@ -58,10 +58,12 @@ private async void OnSave()
                 DedicatedAmount = amountInDecimal
             };
 
-            IsBusy = true;
+           
             await  RestService.SaveTodoItemAsync(newCategory, true);
 
             await Shell.Current.GoToAsync("..");
+
+            IsBusy = true;
```

Comment:
No, I meant before awaits you set `IsBusy = true` and wait for the response to come back and then after awaits you set `IsBusy = false` because you have already got the results and the operation is not busy.